### PR TITLE
Take contigs into account for fixed-size windowing

### LINKED ISSUE #335: Take contigs into account for fixed-size windowing
https://github.com/pystatgen/sgkit/pull/303#discussion_r505711344

### PR DESCRIPTION
**Pull request by tomwhite**
Fixes #335

### CODE REVIEW

File: sgkit/tests/test_window.py [hunk 1]
```diff
@@ -64,13 +64,22 @@ def test_window():
...
+    np.testing.assert_equal(ds[window_contig].values, [0, 0, 0, 0, 0])
     np.testing.assert_equal(ds[window_start].values, [0, 2, 4, 6, 8])
     np.testing.assert_equal(ds[window_stop].values, [2, 4, 6, 8, 10])
 
     with pytest.raises(MergeWarning):
         window(ds, 2, 2)
 
 
+def test_window__multiple_contigs():
+    ds = simulate_genotype_call_dataset(n_variant=15, n_sample=1, n_contig=3)
```
Comment (jeromekelleher) [inline sgkit/tests/test_window.py:76]: It'd be nice to test this with n_contig=n_variant, so we test the degenerate case.

Also be nice to see what happens when n_variant=1.

**Review by jeromekelleher — APPROVED** [review]
LGTM @tomwhite , altough it'd be nice to see some more testing of the edge-cases

File: sgkit/tests/test_window.py (hunk 1 above)
Comment (tomwhite) [reply sgkit/tests/test_window.py]: Added in https://github.com/pystatgen/sgkit/pull/372/commits/399bd045acb7f99f63a91ddce9fae82bf623b280
